tests/Rule/data/ForbidCheckedExceptionInCallableRule/code.php

```diff
@@ -90,15 +99,17 @@ public function test(): void
         $fn4 = function (callable $c) {
             $c(); // implicit throw is ignored (https://github.com/phpstan/phpstan/issues/9779)
         };
+    }
 
-        $this->denied(function () {
-            throw new CheckedException(); // error: Throwing checked exception ForbidCheckedExceptionInCallableRule\CheckedException in closure!
-        });
-
-        $this?->denied(function () {
-            $this->throws(); // error: Throwing checked exception ForbidCheckedExceptionInCallableRule\CheckedException in closure!
-        });
+    public function testExplicitExecution(): void
+    {
+        (function () {
+            throw new CheckedException();
+        })();
```

Comment:
It looks like PHPStan natively understands that this is immediately executed and that it throws, so any line below this in the same method was considered as not executed and did not report any error due to this fact.

Reply:
Not the first time I did this error in tests, pretty tricky. Thx